src/main/scala/org/scanet/strings/StringOps.scala

```diff
@@ -1,14 +1,72 @@
 package org.scanet.strings
 
 import org.scanet.core.TensorType.syntax._
-import org.scanet.core.{Output, TensorType}
+import org.scanet.core.{Output, Shape, TensorType}
 import simulacrum.typeclass
 
 import scala.Ordering.Implicits._
 import scala.language.higherKinds
 
 @typeclass trait StringOps[F[_]] {
 
+  /** Print current tensor during graph evaluation with default destination and formatting options.
+   *
+   * {{{Tensor.vector(1, 2).const.print.eval should be(Tensor.vector(1, 2))}}}
+   *
+   * @return current tensor (with print side effect)
+   */
+  def print[A: TensorType](op: F[A]): F[A] = print(op, PrintOps())
+
+  /** Print current tensor during graph evaluation into specified file with given formatting options.
+   *
+   * {{{Tensor.vector(1, 2).const.print("tensor.log").eval should be(Tensor.vector(1, 2))}}}
+   *
+   * @param file name to write (append) current tensor
+   * @return current tensor (with print side effect)
+   */
+  def print[A: TensorType](op: F[A], file: String): F[A] = print(op, PrintOps(dst = ToFile(file)))
+
+  /** Print current tensor during graph evaluation into specified
+   * output (i.e. stdout or a file) with given formatting options.
+   *
+   * {{{Tensor.vector(1, 2).const.print(dst = LogInfo).eval should be(Tensor.vector(1, 2))}}}
+   *
+   * @param ops output (destination, delimiter) and formatting options
+   * @return current tensor (with print side effect)
+   */
+  def print[A: TensorType](op: F[A], ops: PrintOps): F[A]
```

Comment:
Any chance we can add formating to print like? Otherwise, that will be pretty difficult to use it 
```
learningError.print("Error is: {}")
```

Also, in case we would need to make a composition and print multiple, maybe that would make sense to do something like:
```
val a = 1.const
val b = 2.const
val c = (a puls b) dependsOn ("a + b = ".const join a.format() join b.format())
```

Where `dependsOn` is an identity op

Reply:
cool! that will be much more convenient
turns out format can take multiple Outputs, maybe we can even have something like
```
val c = (a puls b) dependsOn (format("a + b = {} + {}", a, b))
```
another question - in example `print` is missing. Should `dependsOn` or `format` aoutomatically assume `print`? Or should we add it explicitely?

Reply:
or maybe
```
val c = (a puls b) dependsOn (print("a + b = {} + {}", a, b))
```

Reply:
Yep, that was a pseudo-code, missed that, I gues we would need `print` there.

Reply:
Also, we may try to add an additional symbolic name to `dependsOn`, like:
```
val c = (a puls b) <- (format("a + b = {} + {}", a, b))
```

Reply:
```
    val a = 1.const
    val b = 2.const
    val c = (a plus b) << print("a + b = {} + {}", a, b)

    c.eval should be(Tensor.scalar(3))
```
which prints:
![image](https://user-images.githubusercontent.com/5910916/80862874-0887aa80-8c81-11ea-9a8a-43b23229c1e2.png)

Reply:
wow!